--- bluez_simple-agent.orig	2015-05-18 16:02:59.477376771 +0300
+++ bluez_simple-agent	2015-06-06 02:33:25.118006123 +0300
@@ -10,6 +10,8 @@
 import dbus.mainloop.glib
 from optparse import OptionParser
 
+BLUEZ_PIN_CODE = "0987"
+
 def ask(prompt):
 	try:
 		return raw_input(prompt)
@@ -45,7 +47,7 @@
 					in_signature="o", out_signature="s")
 	def RequestPinCode(self, device):
 		print("RequestPinCode (%s)" % (device))
-		return ask("Enter PIN Code: ")
+		return (BLUEZ_PIN_CODE)
 
 	@dbus.service.method("org.bluez.Agent",
 					in_signature="o", out_signature="u")
