--- functions.orig	2019-02-21 19:57:31.902408873 +0300
+++ functions	2019-02-21 20:23:08.865669220 +0300
@@ -55,7 +55,7 @@
 HS_DNS1=${HS_DNS1:-$HS_DNS}
 HS_DNS1=${HS_DNS1:-$(grep '^nameserver' /etc/resolv.conf | head -n1 | awk '{print $2}')}
 if [ $(uname) = "Linux" ]; then
-HS_WANIF=${HS_WANIF:-$(route -n|grep '^0.0.0.0'|head -n1|awk '{print $8}')}
+HS_WANIF=$(/usr/sbin/ip route show | awk '$3 ~ /^[1-9]+/ {print $3}')
 else
 HS_WANIF=${HS_WANIF:-$(route get default | grep interface | awk '{print $2}')}
 fi
@@ -67,12 +67,12 @@
 }
 
 configs1=/tmp/configs1.$$
-addconfig1() { [ -n "$*" ] && cat<<EOF>>$configs1
+addconfig1() { [ -n "$*" ] && cat<<EOF >>$configs1
 $*
 EOF
 }
 configs2=/tmp/configs2.$$
-addconfig2() { [ -n "$*" ] && cat<<EOF>>$configs2
+addconfig2() { [ -n "$*" ] && cat<<EOF >>$configs2
 $*
 EOF
 }
@@ -130,8 +130,8 @@
 
     [ -n "$HS_DHCPGATEWAY" ] && {
 	if [ $(uname) = "Linux" ]; then
-	HS_DHCPRELAYAGENT=${HS_DHCPRELAYAGENT:-$(ifconfig $(route -n|grep '^0.0.0.0'| \
-	awk '{ print $8 }')|grep 'inet addr:'|cut -d: -f 2|cut -d' ' -f1)}
+        # Internet gateway ip address
+	HS_DHCPRELAYAGENT=${HS_DHCPRELAYAGENT:-$(/usr/sbin/ip route show | awk '$3 ~ /^[1-9]+/ {print $3}')}
 	else
 	HS_DHCPRELAYAGENT=${HS_DHCPRELAYAGENT:-$(ifconfig $(netstat -rn | grep 'default' | \
 	head -n1 | awk '{print$7}') | grep 'inet ' | cut -d/ -f 1 | cut -d' ' -f2)}
@@ -382,3 +382,26 @@
     checkfornew
 }
 
+# ---------
+# JambulaTV
+# ---------
+get_default_gateway_dev_and_ip () {
+# Default ethernet device
+NETWORK_ETHERNET_DEVICE=$(/usr/sbin/ip -o -4 link show | awk -F': ' '/mode DEFAULT/ {print $2}' | grep -v lo | head -1)
+#
+# Test to see if there is a default gateway device
+/usr/sbin/ip -o -4 route show | grep -i '^default' > /dev/null 2>&1 && \
+DEFAULT_ROUTE_EXISTS=$?
+#
+if [[ "$DEFAULT_ROUTE_EXISTS" = "0" ]];
+then
+# Default gateway device
+INTERNET_GATEWAY_DEV=$(/usr/sbin/ip -o -4 route show | awk '$3 ~ /^[1-9]+/ {print $5}')
+# Default gateway IP address
+INTERNET_GATEWAY_IP=$(/usr/sbin/ip -o -4 route show | awk '$3 ~ /^[1-9]+/ {print $3}')
+
+else
+INTERNET_GATEWAY_DEV=$NETWORK_ETHERNET_DEVICE
+INTERNET_GATEWAY_IP=$(/usr/sbin/ip -o -4 addr show ${INTERNET_GATEWAY_DEV} | awk -F '[ /]+' '/global/ {print $4}')
+fi
+}
