--- functions.orig	2016-10-27 16:05:47.832262434 +0300
+++ functions	2016-10-27 16:57:54.380144020 +0300
@@ -5,9 +5,6 @@
 # http://coova.github.io/
 #
 
-IPTABLES=/sbin/iptables
-IFCONFIG=/sbin/ifconfig
-
 CHILLI=/etc/chilli
 RUN_D=/var/run
 SBIN=/usr/sbin
@@ -54,7 +51,7 @@
 HS_PAP_OK=${HS_PAP_OK:-on}
 HS_DNS1=${HS_DNS1:-$HS_DNS}
 HS_DNS1=${HS_DNS1:-$(grep '^nameserver' /etc/resolv.conf | head -n1 | awk '{print $2}')}
-HS_WANIF=${HS_WANIF:-$(route -n|grep '^0.0.0.0'|head -n1|awk '{print $8}')}
+HS_WANIF=$(ip route list | grep default | sed -n 's/^.*dev //p' | awk '{print $1}')
 
 bailout() { echo $1; exit; }
 
@@ -63,12 +60,12 @@
 }
 
 configs1=/tmp/configs1.$$
-addconfig1() { [ -n "$*" ] && cat<<EOF>>$configs1
+addconfig1() { [ -n "$*" ] && cat<<EOF >>$configs1
 $*
 EOF
 }
 configs2=/tmp/configs2.$$
-addconfig2() { [ -n "$*" ] && cat<<EOF>>$configs2
+addconfig2() { [ -n "$*" ] && cat<<EOF >>$configs2
 $*
 EOF
 }
@@ -83,7 +80,7 @@
 uamlisten	$HS_UAMLISTEN
 uamport         $HS_UAMPORT
 dhcpif		$HS_LANIF
-uamallowed	"www.coova.org${HS_UAMSERVER:+,$HS_UAMSERVER}$webadmin$uamallow"
+uamallowed	"$HS_UAMSERVER$webadmin$uamallow"
 uamanydns
 $(cat $configs1)
 $HS_RAW_CONFIG1
@@ -125,7 +122,8 @@
     addconfig1 ${HS_GID:+"gid \"$HS_GID\""}
 
     [ -n "$HS_DHCPGATEWAY" ] && {
-	HS_DHCPRELAYAGENT=${HS_DHCPRELAYAGENT:-$(ifconfig $(route -n|grep '^0.0.0.0'|awk '{ print $8 }')|grep 'inet addr:'|cut -d: -f 2|cut -d' ' -f1)}
+        # Internet gateway ip address
+	HS_DHCPRELAYAGENT=${HS_DHCPRELAYAGENT:-$(ip route list | grep default | sed -n 's/^.*via //p' | awk '{print $1}')}
 	addconfig1 "dhcpgateway \"$HS_DHCPGATEWAY\""
 	addconfig1 "dhcprelayagent \"$HS_DHCPRELAYAGENT\""
     }
@@ -367,4 +365,3 @@
     fi
     checkfornew
 }
-
