--- functions.orig	2019-01-03 14:39:05.892281375 +0300
+++ functions		2019-01-21 16:05:40.276025469 +0300
@@ -5,9 +5,6 @@
 # http://coova.github.io/
 #
 
-IPTABLES=/sbin/iptables
-IFCONFIG=/sbin/ifconfig
-
 CHILLI=/etc/chilli
 RUN_D=/var/run
 SBIN=/usr/sbin
@@ -54,7 +51,7 @@
 HS_PAP_OK=${HS_PAP_OK:-on}
 HS_DNS1=${HS_DNS1:-$HS_DNS}
 HS_DNS1=${HS_DNS1:-$(grep '^nameserver' /etc/resolv.conf | head -n1 | awk '{print $2}')}
-HS_WANIF=${HS_WANIF:-$(route -n|grep '^0.0.0.0'|head -n1|awk '{print $8}')}
+HS_WANIF=$(/usr/sbin/ip route show | awk '$3 ~ /^[1-9]+/ {print $3}')
 
 bailout() { echo $1; exit; }
 
@@ -63,12 +60,12 @@
 }
 
 configs1=/tmp/configs1.$$
-addconfig1() { [ -n "$*" ] && cat<<EOF>>$configs1
+addconfig1() { [ -n "$*" ] && cat<<EOF >>$configs1
 $*
 EOF
 }
 configs2=/tmp/configs2.$$
-addconfig2() { [ -n "$*" ] && cat<<EOF>>$configs2
+addconfig2() { [ -n "$*" ] && cat<<EOF >>$configs2
 $*
 EOF
 }
@@ -83,7 +80,7 @@
 uamlisten	$HS_UAMLISTEN
 uamport         $HS_UAMPORT
 dhcpif		$HS_LANIF
-uamallowed	"www.coova.org${HS_UAMSERVER:+,$HS_UAMSERVER}$webadmin$uamallow"
+uamallowed	"$HS_UAMSERVER$webadmin$uamallow"
 uamanydns
 $(cat $configs1)
 $HS_RAW_CONFIG1
@@ -125,7 +122,8 @@
     addconfig1 ${HS_GID:+"gid \"$HS_GID\""}
 
     [ -n "$HS_DHCPGATEWAY" ] && {
-	HS_DHCPRELAYAGENT=${HS_DHCPRELAYAGENT:-$(ifconfig $(route -n|grep '^0.0.0.0'|awk '{ print $8 }')|grep 'inet addr:'|cut -d: -f 2|cut -d' ' -f1)}
+        # Internet gateway ip address
+	HS_DHCPRELAYAGENT=${HS_DHCPRELAYAGENT:-$(/usr/sbin/ip route show | awk '$3 ~ /^[1-9]+/ {print $3}')}
 	addconfig1 "dhcpgateway \"$HS_DHCPGATEWAY\""
 	addconfig1 "dhcprelayagent \"$HS_DHCPRELAYAGENT\""
     }
@@ -368,3 +366,26 @@
     checkfornew
 }
 
+# ---------
+# JambulaTV
+# ---------
+get_default_gateway_dev_and_ip () {
+# Default ethernet device
+NETWORK_ETHERNET_DEVICE=$(/usr/sbin/ip -o -4 link show | awk -F': ' '/mode DEFAULT/ {print $2}' | grep -v lo | head -1)
+#
+# Test to see if there is a default gateway device
+/usr/sbin/ip -o -4 route show | grep -i '^default' > /dev/null 2>&1 && \
+DEFAULT_ROUTE_EXISTS=$?
+#
+if [[ "$DEFAULT_ROUTE_EXISTS" = "0" ]];
+then
+# Default gateway device
+INTERNET_GATEWAY_DEV=$(/usr/sbin/ip -o -4 route show | awk '$3 ~ /^[1-9]+/ {print $5}')
+# Default gateway IP address
+INTERNET_GATEWAY_IP=$(/usr/sbin/ip -o -4 route show | awk '$3 ~ /^[1-9]+/ {print $3}')
+
+else
+INTERNET_GATEWAY_DEV=$NETWORK_ETHERNET_DEVICE
+INTERNET_GATEWAY_IP=$(/usr/sbin/ip -o -4 addr show ${INTERNET_GATEWAY_DEV} | awk -F '[ /]+' '/global/ {print $4}')
+fi
+}
