--- /usr/src/jambulatv/coova-chilli/conf/up.sh	2015-05-28 22:30:24.239887252 +0300
+++ up.sh	2015-06-06 02:43:48.990909126 +0300
@@ -44,10 +44,20 @@
 		    done
 		}
 		
+     	        # Jambula Labs patch to allow UDP ports to be specified in config
+	    	[ -n "HS_UDP_PORTS" ] && {
+		    for port in $HS_UDP_PORTS; do
+	    	        ipt_in -p udp -m udp --dport $port --dst $ADDR -j ACCEPT
+		    done
+    	        }
+ 
 		ipt_in -p udp -d 255.255.255.255 --destination-port 67:68 -j ACCEPT
 		ipt_in -p udp -d $ADDR --destination-port 67:68 -j ACCEPT
 		ipt_in -p udp --dst $ADDR --dport 53 -j ACCEPT
 		ipt_in -p icmp --dst $ADDR -j ACCEPT
+
+                # JambulaTV:  Allow UPNP/DLNA traffic
+		ipt_in -p udp -d 239.255.255.250 --destination-port 1900 -j ACCEPT
 		
 		ipt -A INPUT -i $TUNTAP --dst $ADDR -j DROP
 		
