--- /usr/src/jambulatv/coova-chilli/miniportal/config-local.sh	2015-11-22 08:57:46.247269748 +0300
+++ config-local.sh	2015-12-12 16:18:38.582940347 +0300
@@ -23,8 +23,8 @@
 COOVA_SESSIONID=$CHI_SESSION_ID
 COOVA_CHALLENGE=$CHI_CHALLENGE
 PORTAL_SESSIONID=${COOKIE_PORTAL_SESSIONID:-$SESSIONID}
-FORM_userurl=${FORM_userurl:-http://coova.github.io/}
-    
+FORM_userurl=${FORM_userurl:-$HS_PROVIDER_LINK}
+   
 http_redirect2() {
 cat <<EOF
 HTTP/1.1 302 Redirect
@@ -96,7 +96,72 @@
 
     echo "<table style=\"clear:both;margin:auto;padding-top:10px;\" height=\"30\">
 <tr><td valign=\"center\" align=\"center\" style=\"color:#666;font-size:60%;\">Powered by</td>
-<td valign=\"center\" align=\"center\"><a href=\"http://coova.org/\"><img border=0 src=\"coova.jpg\"></a>
+<td valign=\"center\" align=\"center\"><a href=\"$HS_PROVIDER_LINK\"><img border=0 src=\"coova.jpg\"></a>
+</td></tr></table></body></html>"
+}
+
+jambula_detect_device() {
+/usr/bin/journalctl -r -o cat --no-pager | grep -m 1 -i User-Agent | while read LINE
+do
+	case $LINE in
+		(*iPhone*|*iPad*|*iPod*)
+		USER_AGENT="iOS"
+		;;
+		*Version*Chrome*)
+		USER_AGENT="WebViewer"
+		;;
+		*Android*|*Meego*)
+		USER_AGENT="Android"
+		;;
+		*BlackBerry*)
+		USER_AGENT="Blackberry"
+		;;
+		*Opera*Mini*)
+		USER_AGENT="OperaMini"
+		;;
+		*IEMobile*)
+		USER_AGENT="WinMobile"
+		;;
+		*)
+		USER_AGENT="Other"
+		;;
+	esac	
+		echo "$USER_AGENT" > /tmp/chilli_uagent
+done
+# Set detected device variable
+export DETECTED_DEVICE=$(cat /tmp/chilli_uagent)
+}
+
+jambula_header() {
+    echo "<html>"
+    echo "<head>"
+    uamfile title 0
+
+    echo "<meta http-equiv=\"Cache-control\" content=\"no-cache\"/>"
+    echo "<meta http-equiv=\"Pragma\" content=\"no-cache\"/>"
+    echo "<meta name=\"viewport\" content=\"width=device-width, initial-scale=1\">"
+    echo "<style>"
+    uamfile "landing/css/w3" 0 
+    echo "</style>"
+
+    echo "<script>"
+    uamfile "js" 0 
+    echo "</script>"
+    echo "$1</head>"
+
+    echo "<body$2>"
+    # Put header here
+    echo "<div id=\"body\">"
+}
+
+jambula_footer() {
+    echo "</div>" 
+
+    uamfile "footer" 1 
+
+    echo "<table style=\"clear:both;margin:auto;padding-top:10px;\" height=\"30\">
+<tr><td valign=\"center\" align=\"center\" style=\"color:#666;font-size:60%;\">Powered by</td>
+<td valign=\"center\" align=\"center\"><a href=\"$HS_PROVIDER_LINK\"><img border=0 src=\"coova.jpg\"></a>
 </td></tr></table></body></html>"
 }
 
@@ -212,11 +277,17 @@
     (uamfile "$file" 0
 	echo
 	echo "-------------------------------------------------"
-	echo "Powered by Coova - http://coova.github.io/"
+	echo "Powered by $HS_PROVIDER - $HS_PROVIDER_LINK"
 	echo) | /usr/sbin/sendmail -t -f "$from" && return 0
     return 1;
 }
 
+jambula_check_dns() {
+REMOTE_DOMAIN_NAME=google.com
+host -W 1 $REMOTE_DOMAIN_NAME 208.67.222.222 > /dev/null 2>&1
+DNS_STATUS=$?
+}
+
 FORM_username="${FORM_username:-$FORM_UserName}"
 FORM_username="${FORM_username:-$FORM_Username}"
 FORM_password="${FORM_password:-$FORM_Password}"
