--- config-local.sh.orig	2019-01-03 14:39:06.291281375 +0300
+++ config-local.sh		2019-01-25 15:42:36.099426783 +0300
@@ -23,8 +23,14 @@
 COOVA_SESSIONID=$CHI_SESSION_ID
 COOVA_CHALLENGE=$CHI_CHALLENGE
 PORTAL_SESSIONID=${COOKIE_PORTAL_SESSIONID:-$SESSIONID}
-FORM_userurl=${FORM_userurl:-http://coova.github.io/}
-    
+FORM_userurl=${FORM_userurl:-$HS_PROVIDER_LINK}
+
+FREERADIUS_DB_NAME=COOVACHILLI_DB_NAME
+FREERADIUS_DB_USER=COOVACHILLI_DB_USER
+FREERADIUS_DB_PASS=COOVACHILLI_DB_PASS
+MY_WIFI_DEVICES_FILE=PROJECT_SYSTEM_CONF_DIR/my-wifi-devices.cfg
+
+
 http_redirect2() {
 cat <<EOF
 HTTP/1.1 302 Redirect
@@ -96,7 +102,73 @@
 
     echo "<table style=\"clear:both;margin:auto;padding-top:10px;\" height=\"30\">
 <tr><td valign=\"center\" align=\"center\" style=\"color:#666;font-size:60%;\">Powered by</td>
-<td valign=\"center\" align=\"center\"><a href=\"http://coova.org/\"><img border=0 src=\"coova.jpg\"></a>
+<td valign=\"center\" align=\"center\"><a href=\"$HS_PROVIDER_LINK\"><img border=0 src=\"coova.jpg\"></a>
+</td></tr></table></body></html>"
+}
+
+jambula_detect_device() {
+/usr/bin/journalctl -r -o cat --no-pager | grep -m 1 -i User-Agent | while read LINE
+do
+	case $LINE in
+		*iPhone*|*iPad*|*iPod*)
+		USER_AGENT="iOS"
+		;;
+		*Version*Chrome*)
+		USER_AGENT="WebViewer"
+		;;
+		*Android*|*Meego*)
+		USER_AGENT="Android"
+		;;
+		*BlackBerry*)
+		USER_AGENT="Blackberry"
+		;;
+		*Opera*Mini*)
+		USER_AGENT="OperaMini"
+		;;
+		*IEMobile*)
+		USER_AGENT="WinMobile"
+		;;
+		*)
+		USER_AGENT="Other"
+		;;
+	esac	
+		# Copy detected UA string to temp file 
+		echo "$USER_AGENT" > /tmp/chilli_uagent
+done
+# Set detected device variable
+export DETECTED_DEVICE=$(cat /tmp/chilli_uagent)
+}
+
+jambula_header() {
+    echo "<html>"
+    echo "<head>"
+    uamfile title 0
+
+    echo "<meta http-equiv=\"Cache-control\" content=\"no-cache\"/>"
+    echo "<meta http-equiv=\"Pragma\" content=\"no-cache\"/>"
+    echo "<meta name=\"viewport\" content=\"width=device-width, initial-scale=1\">"
+    echo "<style>"
+    uamfile "landing/css/w3" 0 
+    echo "</style>"
+
+    echo "<script>"
+    uamfile "js" 0 
+    echo "</script>"
+    echo "$1</head>"
+
+    echo "<body$2>"
+    # Put header here
+    echo "<div id=\"body\">"
+}
+
+jambula_footer() {
+    echo "</div>" 
+
+    uamfile "footer" 1 
+
+    echo "<table style=\"clear:both;margin:auto;padding-top:10px;\" height=\"30\">
+<tr><td valign=\"center\" align=\"center\" style=\"color:#666;font-size:60%;\">Powered by</td>
+<td valign=\"center\" align=\"center\"><a href=\"$HS_PROVIDER_LINK\"><img border=0 src=\"coova.jpg\"></a>
 </td></tr></table></body></html>"
 }
 
@@ -212,11 +284,36 @@
     (uamfile "$file" 0
 	echo
 	echo "-------------------------------------------------"
-	echo "Powered by Coova - http://coova.github.io/"
+	echo "Powered by $HS_PROVIDER - $HS_PROVIDER_LINK"
 	echo) | /usr/sbin/sendmail -t -f "$from" && return 0
     return 1;
 }
 
+jambula_check_dns() {
+REMOTE_DOMAIN_NAME=google.com
+SYSTEM_DNS_SERVER=$(sed '/127.0.0.1/d' /etc/resolv.conf | sed 's/^nameserver //g' | sed '/^[a-z]/d' | sed '/^[A-Z]/d' | sed '/^#/d' | sed '/ /d' | uniq | awk 'NR == 1')
+host -W 1 $REMOTE_DOMAIN_NAME $SYSTEM_DNS_SERVER > /dev/null 2>&1
+DNS_STATUS=$?
+}
+
+jambula_add_legal_device () {
+# Check if remote MAC address is in My-WiFi devices file
+grep -iw $REMOTE_MAC $MY_WIFI_DEVICES_FILE > /dev/null 2>&1
+MAC_EXISTS_FILE=$?
+#
+# Check if remote MAC Address already exists in radius MySQL database
+MAC_EXISTS_DB=$(echo "SELECT COUNT(*) FROM radcheck WHERE username = '$REMOTE_MAC';" | mysql -s -u $FREERADIUS_DB_USER -p$FREERADIUS_DB_PASS $FREERADIUS_DB_NAME)
+#
+# If user's MAC address exists in My-WiFI devices file, but does not exist in MySQL DB 
+if [[ "$MAC_EXISTS_FILE" = 0 && "$MAC_EXISTS_DB" != 0 ]];
+then
+# Add MAC Address of device that will ByPass Authentication 
+echo "Adding $REMOTE_MAC to home and authenticated user database  ..."
+#
+echo "INSERT INTO radcheck (UserName, Attribute, Op, Value) VALUES ('$REMOTE_MAC', 'Auth-Type', ':=', 'Accept');" | mysql -u $FREERADIUS_DB_USER -p$FREERADIUS_DB_PASS $FREERADIUS_DB_NAME
+fi
+}
+
 FORM_username="${FORM_username:-$FORM_UserName}"
 FORM_username="${FORM_username:-$FORM_Username}"
 FORM_password="${FORM_password:-$FORM_Password}"
