--- /usr/src/jambulatv/coova-chilli/miniportal/login.chi	2014-07-23 19:54:34.790232793 +0300
+++ login.chi	2014-07-24 12:39:28.750326336 +0300
@@ -14,35 +14,44 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+
+
 . ./config.sh
 
+
+# Check challenge
 if [ -n "$FORM_username" -a -n "$COOVA_CHALLENGE" ]
 then
     http_header
     dologin
+
 else
-    
-    if [ -e "splash.chi" ]
-    then
-	http_redirect "/www/splash.chi"
-    fi
-    
-    http_header
-    
+    http_header   
+
     case "$FORM_res" in
 	success|already)
 	    url=$FORM_redirurl
 	    url=${url:-$FORM_userurl}
-	    header "<meta http-equiv=\"refresh\" content=\"5;url=${url}\"/>"
-	    uamfile "login_success" 1
-	    footer
+
+            check_dns_jambula
+	    # Check if Internet is ON
+	    if [ "$DNS_STATUS" = "0" ];
+		then
+	    # Internet On
+	    header_jambula "<meta http-equiv=\"refresh\" content=\"30;url=${url}\"/>"
+    	    uamfile "post_login_internet_on" 1
+	    exit
+
+		else
+	    # Internet off
+	    header_jambula "<meta http-equiv=\"refresh\" content=\"600;url=${url}\"/>"
+    	    uamfile "post_login_internet_off" 1
 	    exit
+	    fi
 	    ;;
     esac
     
-    header
-    
-    uamfile "login" 1 
+    # Go to top of this file again
     
     if [ "$FORM_res" =  "failed" ]
     then
@@ -54,14 +63,7 @@
 	fi
     fi
     
-    loginform 
-    
-    uamfile "login_footer" 1 
-    [ "$HS_REG_MODE" = "self" ] && uamfile "login_register" 1 
-    [ "$HS_USE_MAP" = "on" ] && uamfile "login_map" 1 
+    	    uamfile "pre_login_portal" 1
     
-    footer
-    
-fi
-
+fi    
 ?>
