--- /usr/src/jambulatv/coova-chilli/miniportal/login.chi	2015-11-22 08:57:46.119261735 +0300
+++ /usr/share/JambulaTV/html/coova/www/login.chi	2016-05-05 14:15:38.508962795 +0300
@@ -14,35 +14,50 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+
+
 . ./config.sh
 
+# Detect device platform
+jambula_detect_device
+
+# Check challenge
 if [ -n "$FORM_username" -a -n "$COOVA_CHALLENGE" ]
 then
     http_header
     dologin
+
 else
-    
-    if [ -e "splash.chi" ]
-    then
-	http_redirect "/www/splash.chi"
-    fi
-    
-    http_header
-    
+    http_header   
+
     case "$FORM_res" in
 	success|already)
 	    url=$FORM_redirurl
 	    url=${url:-$FORM_userurl}
-	    header "<meta http-equiv=\"refresh\" content=\"5;url=${url}\"/>"
-	    uamfile "login_success" 1
-	    footer
+                 
+            # check for DNS connectivity
+            jambula_check_dns
+
+	    # Check if Internet is ON
+	    if [ "$DNS_STATUS" = "0" ];
+		then
+	    # Internet On
+	    jambula_header "<meta http-equiv=\"refresh\" content=\"30;url=${url}\"/>"
+	    uamfile "landing/internet_on/${DETECTED_DEVICE}" 1
+	    jambula_footer
 	    exit
+
+		else
+	    # Internet off
+	    jambula_header "<meta http-equiv=\"refresh\" content=\"600;url=${url}\"/>"
+	    uamfile "landing/internet_off/${DETECTED_DEVICE}" 1
+	    jambula_footer
+	    exit
+	    fi
 	    ;;
     esac
     
-    header
-    
-    uamfile "login" 1 
+    # Go to top of this file again
     
     if [ "$FORM_res" =  "failed" ]
     then
@@ -53,15 +68,10 @@
 	    error "Username and/or passowrd was not valid"
 	fi
     fi
+ 
+            jambula_header
+    	    uamfile "landing/pre_login" 1
+            jambula_footer
     
-    loginform 
-    
-    uamfile "login_footer" 1 
-    [ "$HS_REG_MODE" = "self" ] && uamfile "login_register" 1 
-    [ "$HS_USE_MAP" = "on" ] && uamfile "login_map" 1 
-    
-    footer
-    
-fi
-
+fi    
 ?>
