--- /usr/src/jambulatv/coova-chilli/conf/functions	2015-11-22 08:57:45.934250154 +0300
+++ /etc/chilli/functions	2016-03-21 12:33:09.627749453 +0300
@@ -54,7 +54,7 @@
 HS_PAP_OK=${HS_PAP_OK:-on}
 HS_DNS1=${HS_DNS1:-$HS_DNS}
 HS_DNS1=${HS_DNS1:-$(grep '^nameserver' /etc/resolv.conf | head -n1 | awk '{print $2}')}
-HS_WANIF=${HS_WANIF:-$(route -n|grep '^0.0.0.0'|head -n1|awk '{print $8}')}
+HS_WANIF=$(ip route list | grep default | sed -n 's/^.*dev //p' | awk {'print $1'})
 
 bailout() { echo $1; exit; }
 
@@ -63,12 +63,12 @@
 }
 
 configs1=/tmp/configs1.$$
-addconfig1() { [ -n "$*" ] && cat<<EOF>>$configs1
+addconfig1() { [ -n "$*" ] && cat<<EOF >>$configs1
 $*
 EOF
 }
 configs2=/tmp/configs2.$$
-addconfig2() { [ -n "$*" ] && cat<<EOF>>$configs2
+addconfig2() { [ -n "$*" ] && cat<<EOF >>$configs2
 $*
 EOF
 }
@@ -83,7 +83,7 @@
 uamlisten	$HS_UAMLISTEN
 uamport         $HS_UAMPORT
 dhcpif		$HS_LANIF
-uamallowed	"www.coova.org${HS_UAMSERVER:+,$HS_UAMSERVER}$webadmin$uamallow"
+uamallowed	"$HS_UAMSERVER$webadmin$uamallow"
 uamanydns
 $(cat $configs1)
 $HS_RAW_CONFIG1
@@ -367,4 +367,3 @@
     fi
     checkfornew
 }
-
