--- ./bin/flexget.orig	2019-01-12 17:49:00.676398787 +0300
+++ ./bin/flexget	2019-01-12 17:52:52.205398787 +0300
@@ -5,18 +5,6 @@
 import sys
 from pkg_resources import load_entry_point
 
-# JambulaTV patch
-# ---------------
-import os, ssl, warnings
-# Add support for unverified SSL connections
-if (not os.environ.get('PYTHONHTTPSVERIFY', '') and
-    getattr(ssl, '_create_unverified_context', None)): 
-    ssl._create_default_https_context = ssl._create_unverified_context
-# Ignore all 'unclosed' warnings e.g. unclosed file or socket
-    warnings.filterwarnings(action="ignore", 
-                            message="unclosed", 
-                            category=ResourceWarning)
-
 if __name__ == '__main__':
     sys.argv[0] = re.sub(r'(-script\.pyw?|\.exe)?$', '', sys.argv[0])
     sys.exit(
