--- ./flexget/components/notify/notify.py.orig	2019-02-08 03:04:05.302889575 +0300
+++ ./flexget/components/notify/notify.py	2019-02-16 01:54:49.451545562 +0300
@@ -63,9 +63,9 @@
                         'type': 'string',
                         'default': '{% if not task.failed and not task.accepted %} Task {{task.name}} did not'
                         ' produce any entries.'
-                        '{% else %} [FlexGet] {{task.name}}:'
-                        '{% if task.failed %} {{task.failed|length}} failed entries.{% endif %}'
-                        '{% if task.accepted %} {{task.accepted|length}} new entries downloaded.{% endif %}'
+                        '{% else %} [JambulaTV] '
+                        '{% if task.failed %} Download(s) Failed - {{task.failed|length}} Entries {% endif %}'
+                        '{% if task.accepted %} Download(s) Queued - {{task.accepted|length}} Entries {% endif %}'
                         '{% endif %}',
                     },
                     'message': {'type': 'string'},
