--- flexget/plugins/operate/include.py	2020-06-06 22:03:15.107063651 +0300
+++ flexget/plugins/operate/include.py	2020-06-08 16:11:26.765939972 +0300
@@ -1,4 +1,5 @@
 import os
+import io
 
 import yaml
 from loguru import logger
@@ -41,6 +42,14 @@
             with open(file, encoding='utf-8') as inc_file:
                 include = yaml.safe_load(inc_file)
                 inc_file.flush()
+            #
+	    # JambulaTV: Check if included file exists
+            if os.path.exists(file):
+                include = yaml.load(io.open(file, encoding='utf-8'), Loader=yaml.FullLoader)
+            else:
+                log.warning('JambulaTV: Please add the file: %s ' % file)
+                break
+            #
             errors = process_config(include, plugin.plugin_schemas(interface='task'))
             if errors:
                 logger.error('Included file {} has invalid config:', file)
