--- /tmp/Makefile	2015-03-02 00:04:41.428553154 +0300
+++ /usr/src/jambulatv/hostap/hostapd/Makefile	2015-03-02 00:14:47.388430711 +0300
@@ -10,7 +10,7 @@
 CFLAGS += -I$(abspath ../src)
 CFLAGS += -I$(abspath ../src/utils)
 
-export BINDIR ?= /usr/local/bin/
+export BINDIR ?= /usr/sbin/
 
 # Uncomment following line and set the path to your kernel tree include
 # directory if your C library does not include all header files.
