--- fw_iptables.c.orig		2019-08-20 02:15:48.992287321 +0300
+++ fw_iptables.c		2019-08-20 02:16:31.046780379 +0300
@@ -193,9 +193,9 @@
 
 	for (i = 0; i < 5; i++) {
 		if (fw_quiet) {
-			rc = execute("%s --wait %s > /dev/null 2>&1", iptables, fmt_cmd);
+			rc = execute("%s %s > /dev/null 2>&1", iptables, fmt_cmd);
 		} else {
-			rc = execute("%s --wait %s", iptables, fmt_cmd);
+			rc = execute("%s %s", iptables, fmt_cmd);
 		}
 
 		if (rc == 4) {
