--- unicodeobject.c.orig	2019-03-17 17:41:35.154401817 +0300
+++ unicodeobject.c		2019-03-17 17:44:21.843598641 +0300
@@ -9473,6 +9473,7 @@
     PyObject *thousands_sep,
     Py_UCS4 *maxchar)
 {
+    min_width = Py_MAX(0, min_width);
     if (writer) {
         assert(digits != NULL);
         assert(maxchar == NULL);
@@ -9483,7 +9484,6 @@
     }
     assert(0 <= d_pos);
     assert(0 <= n_digits);
-    assert(0 <= min_width);
     assert(grouping != NULL);
 
     if (digits != NULL) {
