--- sql.orig	2015-05-25 16:50:44.782037579 +0300
+++ sql.good	2015-08-27 20:43:12.650598773 +0300
@@ -28,7 +28,7 @@
 	#    * rlm_sql_sqlite
 	#    * rlm_sql_null (log queries to disk)
 	#
-	driver = "rlm_sql_null"
+	driver = "rlm_sql_mysql"
 
 #
 #	Several drivers accept specific options, to set them, a
@@ -79,14 +79,14 @@
 	#
 	# If you're using rlm_sql_null, then it should be the type of
 	# database the logged queries are going to be executed against.
-	dialect = "sqlite"
+	dialect = "mysql"
 
 	# Connection info:
 	#
-#	server = "localhost"
-#	port = 3306
-#	login = "radius"
-#	password = "radpass"
+	server = "localhost"
+	port = 3306
+	login = "FREERADIUS_DB_USER"
+	password = "FREERADIUS_DB_PASS"
 
 	# Database table configuration for everything except Oracle
 	radius_db = "radius"
@@ -223,7 +223,7 @@
 
 	# Set to 'yes' to read radius clients from the database ('nas' table)
 	# Clients will ONLY be read on server startup.
-#	read_clients = yes
+	read_clients = yes
 
 	# Table to keep radius client info
 	client_table = "nas"
