--- inner-tunnel.orig	2014-08-26 21:19:51.123369325 +0300
+++ inner-tunnel	2014-08-26 21:23:18.700464636 +0300
@@ -129,7 +129,11 @@
 	#  is meant to mirror the "users" file.
 	#
 	#  See "Authorization Queries" in sql.conf
-	-sql
+	sql
+
+	# Jambula Labs - SQL counters
+	chillispot_max_bytes
+	noresetcounter
 
 	#
 	#  If you are using /etc/smbpasswd, and are also doing
@@ -246,7 +250,7 @@
 
 	#
 	#  See "Simultaneous Use Checking Queries" in sql.conf
-#	sql
+	sql
 }
 
 
@@ -269,13 +273,13 @@
 	#  After authenticating the user, do another SQL query.
 	#
 	#  See "Authentication Logging Queries" in sql.conf
-	-sql
+	sql
 
 	#
 	#  Instead of sending the query to the SQL server,
 	#  write it into a log file.
 	#
-#	sql_log
+	#sql_log
 
 	#
 	#  Un-comment the following if you have set
