--- smb.conf.orig	2013-01-31 11:33:02.000000000 +0300
+++ smb.conf	2014-07-24 21:05:44.816904000 +0300
@@ -86,10 +86,11 @@
 # max protocol = used to define the supported protocol. The default is NT1. You
 # can set it to SMB2 if you want experimental SMB2 support.
 #
-	workgroup = MYGROUP
-	server string = Samba Server Version %v
+	workgroup = JAMBULATV
+	server string = My media at JambulaTV
 
-;	netbios name = MYSERVER
+	netbios name = JambulaTV
+        name resolve order = bcast host 
 
 ;	interfaces = lo eth0 192.168.12.2/24 192.168.13.2/24
 ;	hosts allow = 127. 192.168.12. 192.168.13.
@@ -279,42 +280,10 @@
 
 #============================ Share Definitions ==============================
 
-[homes]
-	comment = Home Directories
-	browseable = no
+	[Media-Disk]
+	comment = My Media at JambulaTV
+	path = /JambulaTV
+	browseable = yes
+	public = yes
 	writable = yes
-;	valid users = %S
-;	valid users = MYDOMAIN\%S
-
-[printers]
-	comment = All Printers
-	path = /var/spool/samba
-	browseable = no
-	guest ok = no
-	writable = no
-	printable = yes
-
-# Un-comment the following and create the netlogon directory for Domain Logons:
-;	[netlogon]
-;	comment = Network Logon Service
-;	path = /var/lib/samba/netlogon
-;	guest ok = yes
-;	writable = no
-;	share modes = no
-
-# Un-comment the following to provide a specific roving profile share.
-# The default is to use the user's home directory:
-;	[Profiles]
-;	path = /var/lib/samba/profiles
-;	browseable = no
-;	guest ok = yes
-
-# A publicly accessible directory that is read only, except for users in the
-# "staff" group (which have write permissions):
-;	[public]
-;	comment = Public Stuff
-;	path = /home/samba
-;	public = yes
-;	writable = yes
-;	printable = no
-;	write list = +staff
+	printable = no
