--- sshuttle.orig	2019-08-08 01:16:16.653549311 +0300
+++ sshuttle		2019-08-08 01:18:25.575640615 +0300
@@ -5,6 +5,18 @@
 import sys
 from pkg_resources import load_entry_point
 
+# JambulaTV patch
+# ---------------
+import os, ssl, warnings
+# Add support for unverified SSL connections
+if (not os.environ.get('PYTHONHTTPSVERIFY', '') and
+    getattr(ssl, '_create_unverified_context', None)): 
+    ssl._create_default_https_context = ssl._create_unverified_context
+# Ignore all 'unclosed' warnings e.g. unclosed file or socket
+    warnings.filterwarnings(action="ignore", 
+                            message="unclosed", 
+                            category=ResourceWarning)
+
 if __name__ == '__main__':
     sys.argv[0] = re.sub(r'(-script\.pyw?|\.exe)?$', '', sys.argv[0])
     sys.exit(
