--- env_android.py.orig	2016-11-22 23:13:27.600216822 +0300
+++ env_android.py	2016-11-23 20:45:34.629495519 +0300
@@ -17,10 +17,10 @@
         "YHNtYoIvt5R3X6YZylbPftF/8ayWTALBgcqhkjOOAQDBQADLwAwLAIUAKYCp0d6z4QQdyN74JDfQ2WCyi8CFDUM4CaNB+ceVXd" \
         "KtOrNTQcc0e+t"
 
-    _MD5_CLASSES = "ry9Xz6kVioQctwA3G9z62Q=="
+    _MD5_CLASSES = "14w/wF67XBf2vTc+qALwKQ=="
     _KEY = "eQV5aq/Cg63Gsq1sshN9T3gh+UUp0wIw0xgHYT1bnCjEqOJQKCRrWxdAe2yvsDeCJL+Y4G3PRD2HUF7oUgiGo8vGlNJOaux26k+A2F3hj8A="
 
-    _VERSION = "2.12.556"
+    _VERSION = "2.16.148"
     _OS_NAME = "Android"
     _OS_VERSION = "4.3"
     _DEVICE_NAME = "armani"
