--- yowsup/setup.py.orig	2021-12-26 20:18:46.722048772 +0300
+++ yowsup/setup.py		2022-01-07 14:45:35.929515471 +0300
@@ -5,7 +5,7 @@
 import platform
 import sys
 
-deps = ['consonance==0.1.5', 'argparse', 'python-axolotl==0.2.2', 'six==1.10', 'appdirs', 'protobuf>=3.6.0']
+deps = ['consonance>=0.1.5', 'argparse', 'python-axolotl>=0.2.2', 'six>=1.10', 'appdirs', 'protobuf>=3.6.0']
 
 if sys.version_info < (2, 7):
     deps.append('importlib')
