--- test_tokendictionary.py	2019-05-22 13:15:39.156720950 +0300
+++ test_tokendictionary.py	2020-06-03 23:48:14.448237262 +0300
@@ -5,16 +5,16 @@
         self.tokenDictionary = TokenDictionary()
 
     def test_getToken(self):
-        self.assertEqual(self.tokenDictionary.getToken(80), "iq")
+        self.assertEqual(self.tokenDictionary.getToken(10), "iq")
 
     def test_getIndex(self):
-        self.assertEqual(self.tokenDictionary.getIndex("iq"), (80, False))
+        self.assertEqual(self.tokenDictionary.getIndex("iq"), (10, False))
 
     def test_getSecondaryToken(self):
-        self.assertEqual(self.tokenDictionary.getToken(238), "amrnb")
+        self.assertEqual(self.tokenDictionary.getToken(238), "lc")
 
     def test_getSecondaryTokenExplicit(self):
-        self.assertEqual(self.tokenDictionary.getToken(11, True), "wmv")
+        self.assertEqual(self.tokenDictionary.getToken(11, True), "reject")
 
     def test_getSecondaryIndex(self):
-        self.assertEqual(self.tokenDictionary.getIndex("wmv"), (11, True))
\ No newline at end of file
+        self.assertEqual(self.tokenDictionary.getIndex("reject"), (11, True))
\ No newline at end of file
