--- env_android.py.orig		2019-03-12 23:19:16.016725498 +0300
+++ env_android.py		2019-03-12 23:59:21.013077565 +0300
@@ -17,10 +17,10 @@
         "YHNtYoIvt5R3X6YZylbPftF/8ayWTALBgcqhkjOOAQDBQADLwAwLAIUAKYCp0d6z4QQdyN74JDfQ2WCyi8CFDUM4CaNB+ceVXd" \
         "KtOrNTQcc0e+t"
 
-    _MD5_CLASSES = "JHeEdzB6SUug0Z9nuhGWFQ=="
+    _MD5_CLASSES = "mdQhr/+1g4eNRAa7R0L/hw=="
     _KEY = "eQV5aq/Cg63Gsq1sshN9T3gh+UUp0wIw0xgHYT1bnCjEqOJQKCRrWxdAe2yvsDeCJL+Y4G3PRD2HUF7oUgiGo8vGlNJOaux26k+A2F3hj8A="
 
-    _VERSION = "2.18.162"
+    _VERSION = "2.19.68"
     _OS_NAME = "Android"
 	#new
     _OS_VERSION = "4.3"
