--- dvb_frontend.c.orig	2016-10-07 20:00:36.386668770 +0300
+++ dvb_frontend.c	2016-10-07 20:02:41.564005866 +0300
@@ -51,7 +51,7 @@
 static int dvb_shutdown_timeout;
 static int dvb_force_auto_inversion;
 static int dvb_override_tune_delay;
-static int dvb_powerdown_on_sleep = 1;
+static int dvb_powerdown_on_sleep = 0;
 static int dvb_mfe_wait_time = 5;
 
 module_param_named(frontend_debug, dvb_frontend_debug, int, 0644);
